packages/valory/skills/decision_maker_abci/behaviours/sell_outcome_token.py

```diff
@@ -0,0 +1,215 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+
+"""This module contains the behaviour for selling a token."""
+from typing import Any, Generator, Optional, cast
+
+from aea.configurations.data_types import PublicId
+from hexbytes import HexBytes
+
+from packages.valory.contracts.conditional_tokens.contract import (
+    ConditionalTokensContract,
+)
+from packages.valory.contracts.erc20.contract import ERC20
+from packages.valory.contracts.market_maker.contract import (
+    FixedProductMarketMakerContract,
+)
+from packages.valory.protocols.contract_api import ContractApiMessage
+from packages.valory.skills.decision_maker_abci.behaviours.base import (
+    DecisionMakerBaseBehaviour,
+    WaitableConditionType,
+    remove_fraction_wei,
+)
+from packages.valory.skills.decision_maker_abci.models import MultisendBatch
+from packages.valory.skills.decision_maker_abci.payloads import MultisigTxPayload
+from packages.valory.skills.decision_maker_abci.states.sell_outcome_token import (
+    SellOutcomeTokenRound,
+)
+
+
+class SellTokenBehaviour(DecisionMakerBaseBehaviour):
+    """A behaviour in which the agents sell a token."""
+
+    matching_round = SellOutcomeTokenRound
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the sell token behaviour."""
+        super().__init__(**kwargs)
+        self.sell_amount = 0
+        self.outcome_token_balance = 0
+        self.return_amount = 0
+
+    @property
+    def market_maker_contract_address(self) -> str:
+        """Get the contract address of the market maker on which the service is going to place the bet."""
+        return self.sampled_bet.id
+
+    @property
+    def outcome_index(self) -> int:
+        """Get the index of the outcome for which the service is going to sell token."""
+        return cast(int, self.synchronized_data.vote)
+
+    @property
+    def current_condition_id(self) -> PublicId:
+        """Get the current condition id."""
+        return self.shared_state.current_condition_id
+
+    def check_outcome_token_balance(self) -> WaitableConditionType:
+        """Check the safe's balance for a particular outcome token."""
+        response_msg = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+            contract_address=self.params.conditional_tokens_address,
+            contract_public_id=ConditionalTokensContract.contract_id,
+            contract_callable="balanceOf",
+            data_key="data",
+            placeholder="outcome_token_balance",
+            id=self.current_condition_id,
+            owner=self.synchronized_data.safe_contract_address,
+        )
+        return response_msg
```

Comment:
You could use the `check_balance` from the base behaviour instead.

Reply:
no longer needed as the sell amount is now taken from the `bet.invested_amount`